type-c-service/src/wrapper/cfu.rs

```diff
@@ -0,0 +1,281 @@
+//! CFU message bridge
+//! TODO: remove this once we have a more generic FW update implementation
+use embassy_futures::select::{select, Either};
+use embedded_cfu_protocol::protocol_definitions::*;
+use embedded_services::cfu::component::{InternalResponseData, RequestData};
+use embedded_services::type_c::controller::Controller;
+use embedded_services::{debug, error};
+
+use super::*;
+
+/// Current state of the firmware update process
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FwUpdateState {
+    /// None in progress
+    Idle,
+    /// Firmware update in progress
+    /// Contains number of ticks [`super::DEFAULT_FW_UPDATE_TICK_INTERVAL_MS`] that have passed
+    InProgress(u8),
+    /// Firmware update has failed and the device is in an unknown state
+    Recovery,
+}
+
+impl FwUpdateState {
+    /// Check if the firmware update is in progress
+    pub fn in_progress(&self) -> bool {
+        matches!(self, FwUpdateState::InProgress(_) | FwUpdateState::Recovery)
+    }
+}
+
+impl<const N: usize, C: Controller, V: FwVersionValidator> ControllerWrapper<'_, N, C, V> {
+    /// Process a GetFwVersion command
+    async fn process_get_fw_version(&self, target: &mut C) -> InternalResponseData {
+        let version = match target.get_active_fw_version().await {
+            Ok(v) => v,
+            Err(Error::Pd(e)) => {
+                error!("Failed to get active firmware version: {:?}", e);
+                return InternalResponseData::ComponentBusy;
+            }
+            Err(Error::Bus(_)) => {
+                error!("Failed to get active firmware version, bus error");
+                return InternalResponseData::ComponentBusy;
+            }
+        };
+
+        let dev_inf = FwVerComponentInfo::new(FwVersion::new(version), self.cfu_device.component_id());
+        let comp_info: [FwVerComponentInfo; MAX_CMPT_COUNT] = [dev_inf; MAX_CMPT_COUNT];
+        InternalResponseData::FwVersionResponse(GetFwVersionResponse {
+            header: GetFwVersionResponseHeader::new(1, GetFwVerRespHeaderByte3::NoSpecialFlags),
+            component_info: comp_info,
+        })
+    }
+
+    /// Process a GiveOffer command
+    async fn process_give_offer(&self, target: &mut C, offer: &FwUpdateOffer) -> Option<InternalResponseData> {
+        if offer.component_info.component_id != self.cfu_device.component_id() {
+            return None;
+        }
+
+        let version = match target.get_active_fw_version().await {
+            Ok(v) => v,
+            Err(Error::Pd(e)) => {
+                error!("Failed to get active firmware version: {:?}", e);
+                return Some(InternalResponseData::ComponentBusy);
```

Comment:
This error doesn't indicate the underlying cause in any way, making debugging more difficult. Can you wrap the inner `Error::Pd(e)`?

Same suggestion for other indiscriminant uses of `ComponentBusy`